mlfinlab/clustering/onc.py

```diff
@@ -55,8 +55,7 @@ def _cluster_kmeans_base(corr_mat: pd.DataFrame, max_num_clusters: int = 10, rep
     """
 
     # Distance matrix
-    distance = ((1 - corr_mat.fillna(0)) / 2.0) ** 0.5
-
+    distance = (((1 - corr_mat.fillna(0)) / 2.0) ** 0.5).fillna(0)
```

Comment:
@proskurin are we sure we should do this?

Reply:
I don't think so... It would mean that values that return NA would be very close in distance. I really think this should be removed and tested before we add it.

Reply:
When I tested the ONC  sklearn's K-Means algorithm returns error if a Nan value goes through.

Reply:
Under what circumstances does a NaN value get passed?

Reply:
I have done tests and it confirms that the problem is not with the ONC scripts distance matrix but it occurs only when information theory matrix are used as dependence matrix. There is no error with linear correlation. So I remove the fillna(0) and reverted it back.